core/storage/translation/gae_models.py

```diff
@@ -34,6 +35,160 @@
+    def get_deletion_policy() -> base_models.DELETION_POLICY:
+        """Model doesn't contain any data directly corresponding to a user."""
+        return base_models.DELETION_POLICY.NOT_APPLICABLE
+
+    @staticmethod
+    def get_model_association_to_user(
+    ) -> base_models.MODEL_ASSOCIATION_TO_USER:
+        """Model does not contain user data."""
+        return base_models.MODEL_ASSOCIATION_TO_USER.NOT_CORRESPONDING_TO_USER
+
+    @classmethod
+    def get_export_policy(cls) -> Dict[str, base_models.EXPORT_POLICY]:
+        """Model doesn't contain any data directly corresponding to a user."""
+        return dict(super(cls, cls).get_export_policy(), **{
+            'entity_id': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'entity_type': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'entity_version': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'language_code': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'translations': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+        })
+
+    @staticmethod
+    def _generate_id(
+        entity_type: str,
+        entity_id: str,
+        entity_version: int,
+        language_code: str
+    ) -> str:
+        """The method use to generate unique id for entity translations model.
+
+        Args:
+            entity_type: str. Type of an entity.
+            entity_id: str. Id of an entity.
+            entity_version: int. Version of an entity.
+            language_code: str. Language code for a given entity.
+
+        Returns:
+            str. Returns a unique id of the form
+            [entity_type]-[entity_id]-[entity_version]-[language_code].
+        """
+        return '%s-%s-%s-%s' % (
+            entity_type, entity_id, entity_version, language_code)
+
+    @classmethod
+    def get_model(
+        cls,
+        entity_type: str,
+        entity_id: str,
+        entity_version: int,
+        language_code: str
+    ) -> EntityTranslationsModel:
+        """Gets EntityTranslationsModel by help of entity_type, entity_id,
+        entity_version and language_code.
+
+        Args:
+            entity_type: str. Type of an entity.
+            entity_id: str. Id of an entity.
+            entity_version: int. Version of an entity.
+            language_code: str. Language code for a given entity.
+
+        Returns:
+            EntityTranslationsModel. The EntityTranslationsModel
+            instance corresponding to the given inputs, if such a translation
+            exists, or None if no translation is found.
+        """
+        model_id = cls._generate_id(
+            entity_type, entity_id, entity_version, language_code)
+        return cls.get_by_id(model_id)
+
+    @classmethod
+    def get_all_for_entity(
+        cls,
+        entity_type: str,
+        entity_id: str,
+        entity_version: int
+    ) -> Sequence[Optional[EntityTranslationsModel]]:
+        """Gets EntityTranslationsModels by help of entity_type, entity_id,
```

Comment:
Gets EntityTranslationsModels corresponding to the given entity, for all languages in which such models exist.

Reply:
Updated.